source/material_model/depth_dependent.cc

```diff
@@ -0,0 +1,444 @@
+      int i=0;
+      const int nlayers = depth_values.size()-1;
+      while ( depth > depth_values[i] && i<nlayers ) i++; /* incerment i until depth is above base of layer i */
+      return viscosity_values[i];
+    }
+
+    template <int dim>
+    void
+    DepthDependent<dim>::initialize()
+    {
+      /* If using the File method for depth-dependence, initialize the lookup table */
+      if ( viscosity_method == File )
+        {
+          radial_viscosity_lookup.reset(new internal::RadialViscosityLookup(datadirectory+radial_viscosity_file_name));
+        }
+    }
+
+    template <int dim>
+    double
+    DepthDependent<dim>::calculate_depth_dependent_prefactor(const double &depth) const
+    {
+      /* The depth dependent prefactor is the multiplicative factor by which the
+       * viscosity computed by the base model's evaluate mathod will be scaled */
+      const double reference_viscosity = base_model->reference_viscosity();
+      if ( viscosity_method == File )
+        {
+          return radial_viscosity_lookup->radial_viscosity(depth)/reference_viscosity;
+        }
+      else if ( viscosity_method == Function )
+        {
+          Point<1> dpoint(depth);
+          const double viscosity_depth_prefactor = viscosity_function.value(dpoint);
+          Assert (viscosity_depth_prefactor > 0.0, ExcMessage("Viscosity depth function should be larger than zero"));
+          return viscosity_depth_prefactor/reference_viscosity;
+        }
+      else if (viscosity_method == List)
+        {
+          return viscosity_from_list(depth)/reference_viscosity;
+        }
+      else if (viscosity_method == None)
+        {
+          return 1.0;
+        }
+      else
+        {
+          Assert( false, ExcMessage("Invalid method for viscosity depth dependence") );
+          return 0.0;
+        }
+    }
+
+
+    template <int dim>
+    void
+    DepthDependent<dim>::evaluate(const typename Interface<dim>::MaterialModelInputs &in,
+                                  typename Interface<dim>::MaterialModelOutputs &out) const
+    {
+      base_model -> evaluate(in,out);
+      /* Scale the base model viscosity value by the depth dependenct prefactor */
+      for (unsigned int i=0; i < out.viscosities.size(); ++i)
+        {
+          const double depth = this->get_geometry_model().depth(in.position[i]);
+          out.viscosities[i] *= calculate_depth_dependent_prefactor( depth );
+        }
+    }
+
+
+    template <int dim>
+    void
+    DepthDependent<dim>::declare_parameters (ParameterHandler &prm)
+    {
+      prm.enter_subsection("Material model");
+      {
+        prm.enter_subsection("Depth dependent model");
+        {
+          prm.declare_entry("Base model","simple",
+                            Patterns::Selection("simple|simpler|simple compressible|Steinberger|table|composition reaction|Morency and Doin|latent heat|latent heat melt|multicomponent"),
```

Comment:
if you rebase to master, you can use the function in namespace MaterialModel to get at this list now.